### DISCUSSION TITLE
Clarification

### DISCUSSION THREAD

**Discussion by aelhadee** [named-data/python-ndn#59] [General]
Hello, 
I am trying to implement a sender and receiver using python-ndn and currently the receiver sends the interest packet in order for the sender to send the data. In order to send the data faster, the receiver will have to send the interest data faster and more frequently. 

I was wondering if there is a configuration or a way to reduce the number and frequency of interest packets that the receiver has to send to receiver the data and still receive the data at a fast rate? 

For example, if I am sending data every 10 milliseconds and that means the receiver will send an interest packet every 10 milliseconds. So, is it possible for the receiver to send an interest packet every, let's say, 500 ms and still get the data packets every 10 ms? The main reason is to reduce the load on the network.

Thanks,
AE

**Comment by zjkmxy** [comment]
Hello, there is no such setting. NDN traffic control requires that every Data must be corresponding to exactly one Interest. Therefore, your design is not supported by NDN.
Please refer to [Design Principles](https://named-data.net/project/ndn-design-principles/) No. 6. You may also send an email to NDN mailing list for detailed reasons.

**Reply by aelhadee** [reply]
Thank you for the clarification. 

Another question - My understanding that security is enabled by default and each data packet being sent has an ECDSA signature attached to it. Is there a way to temporarily disable security and not attach signatures to data packets?

**Reply by zjkmxy** [reply]
There is a [NullSigner](https://github.com/named-data/python-ndn/blob/master/src/ndn/security/signer/null_signer.py). This can be set in `prepare_data` or `put_data` function via [Keyword Arguments](https://python-ndn.readthedocs.io/en/latest/src/app.html#signature). This signer will gives an zero-size SignatureValue.
An alternative option is `no_signature=True`, which will remove SignatureInfo and SignatureValue in the data. But NFD may drop your packet if you do so.

**Reply by aelhadee** [reply]
Thanks a lot, that helped!
